### PR TITLE
Update release notes action for new release model

### PR DESCRIPTION
**Pull request by jpelgrom**
## Summary

The new release model #3597 dropped tags on every release + `beta` in tags for beta releases. Switch to GitHub generated release notes to get them back + consistency with 'normal' releases.

(this PR is #3656 in this repo instead to be able to change permissions which is required for it to work)

## Screenshots

n/a

## Link to pull request in Documentation repository

n/a

### CODE REVIEW

**Comment by jpelgrom** [comment]
Getting closer but it is including filenames for unclear reasons

File: .github/actions/create-release-notes/action.yml [hunk 1]
```diff
@@ -9,15 +16,30 @@ runs:
       run: |
         mkdir -p ./app/build/outputs/
         
-        echo "Previous Release Commit:"
-        git tag -l --sort=-creatordate | grep -v beta- | head -n 1
-        previous=`git tag -l --sort=-creatordate | grep -v beta- | head -n 1`
+        echo "Previous Release Tag:"
+        TAG=`\
+          curl --no-progress-meter --request GET \
+            --url "https://api.github.com/repos/home-assistant/android/releases/latest" \
```
Comment (JBassett) [inline .github/actions/create-release-notes/action.yml:22]: Docs: https://octokit.github.io/rest.js/v19
`Get the latest release`

File: .github/actions/create-release-notes/action.yml [hunk 2]
```diff
@@ -9,15 +16,30 @@ runs:
...
+        `
+        echo $TAG
         
-        echo "# Changes:" > ./app/build/outputs/changelogGithub
-        echo "Changlog:"
-        git log --format="* %s" ${previous}..HEAD
-        git log --format="* %s" ${previous}..HEAD >> ./app/build/outputs/changelogGithub
+        echo "Full Changelog:"
+        CHANGELOG=`\
+          curl --no-progress-meter --request POST \
```
[the remark this answers is not in the document]
Comment (jpelgrom) [reply .github/actions/create-release-notes/action.yml:32]: I was not aware of this, searching for actions that can generate release notes I only found 3rd party solutions. I'll see if that is easier to integrate.

[the remark this answers is not in the document]
Comment (jpelgrom) [reply .github/actions/create-release-notes/action.yml]: Done, thanks for pointing it out :)

**Comment by jpelgrom** [comment]
Output now looks like expected:

```
Full Changelog:
## What's Changed
* Implement a frontend-like single entity picker by @jpelgrom in https://github.com/home-assistant/android/pull/3653
* Hide android auto sensor section from devices that do not support it by @dshokouhi in https://github.com/home-assistant/android/pull/3657
* Bump io.sentry:sentry-android from 6.25.0 to 6.25.1 by @dependabot in https://github.com/home-assistant/android/pull/3658
* Update change log week of 7/14 by @dshokouhi in https://github.com/home-assistant/android/pull/3660
* Add CarInfo sensors by @drosoCode in https://github.com/home-assistant/android/pull/3399
* Update/replace deprecations in app module by @jpelgrom in https://github.com/home-assistant/android/pull/3661
* Use new entity picker for more settings by @jpelgrom in https://github.com/home-assistant/android/pull/3662
* Add toggle for Assist as assistant app on Wear by @jpelgrom in https://github.com/home-assistant/android/pull/3663
* Breaking change: support last_used/preferred pipeline IDs and use last_used by default by @jpelgrom in https://github.com/home-assistant/android/pull/3669
* Update Thread network sync to update/delete HA networks by @jpelgrom in https://github.com/home-assistant/android/pull/3676
* Bump io.sentry:sentry-android from 6.25.1 to 6.25.2 by @dependabot in https://github.com/home-assistant/android/pull/3672
* Hilt Dependency Update by @JBassett in https://github.com/home-assistant/android/pull/3681
* Send state based on reported status, apply car prefix to car sensors, add some logging around new car sensors by @dshokouhi in https://github.com/home-assistant/android/pull/3668
* Bump io.sentry:sentry-android from 6.25.2 to 6.26.0 by @dependabot in https://github.com/home-assistant/android/pull/3682
* Only show unit of measurement if we have it and if the state is not a string while managing sensors by @dshokouhi in https://github.com/home-assistant/android/pull/3685
* Don't send group summary notifications to Android Auto by @jpelgrom in https://github.com/home-assistant/android/pull/3686
* Only add up to the allowed limit of list items to android auto by @dshokouhi in https://github.com/home-assistant/android/pull/3687
* Add sensors to report car energy profile by @dshokouhi in https://github.com/home-assistant/android/pull/3688
* Add favorites selection for Android Auto by @dshokouhi in https://github.com/home-assistant/android/pull/3670
* Resume Assist voice input on new intent (as assistant app/brought to front) by @jpelgrom in https://github.com/home-assistant/android/pull/3692
* Text field consistency updates by @jpelgrom in https://github.com/home-assistant/android/pull/3693
* Update change log week of 7/21 by @dshokouhi in https://github.com/home-assistant/android/pull/3691
* Match android auto dependency versions by @dshokouhi in https://github.com/home-assistant/android/pull/3695
* Update/replace deprecations in common module by @jpelgrom in https://github.com/home-assistant/android/pull/3698
* Update/replace deprecations and unused code in wear module by @jpelgrom in https://github.com/home-assistant/android/pull/3699
* Show application names, not only package names, in sensor settings by @jpelgrom in https://github.com/home-assistant/android/pull/3700
* Use WifiManager deprecations again for instant 'location' info by @jpelgrom in https://github.com/home-assistant/android/pull/3705
* Fix Wear more details crash with Kelvin light temperature by @jpelgrom in https://github.com/home-assistant/android/pull/3706
* Show measurement unit in Wear OS complication by @slovdahl in https://github.com/home-assistant/android/pull/3703
* Bump actions/setup-java from 3.11.0 to 3.12.0 by @dependabot in https://github.com/home-assistant/android/pull/3707
* Android auto: Switch to GridTemplate for home screen and default to favorites if defined by @dshokouhi in https://github.com/home-assistant/android/pull/3694
* Fix version code for F-Droid builds by @jpelgrom in https://github.com/home-assistant/android/pull/3717
* Bump some project dependencies by @dshokouhi in https://github.com/home-assistant/android/pull/3718
* Add navigation entities to favorites and convert nav screen to grid by @dshokouhi in https://github.com/home-assistant/android/pull/3725
* Update sensors when driving interface is open or resumed by @dshokouhi in https://github.com/home-assistant/android/pull/3726
* Fix old server references in app-level preferences by @jpelgrom in https://github.com/home-assistant/android/pull/3728
* Show favorites only if selected server has favorites by @dshokouhi in https://github.com/home-assistant/android/pull/3729
* Smarter discovery for already added instances by @jpelgrom in https://github.com/home-assistant/android/pull/3723
* Update change log week of 7/28 by @dshokouhi in https://github.com/home-assistant/android/pull/3730
* First Steps into making Gradle Catalogs by @JBassett in https://github.com/home-assistant/android/pull/3724
* Bump kotlinx-coroutines from 1.7.2 to 1.7.3 by @dependabot in https://github.com/home-assistant/android/pull/3735
* Implement support for multiple Shortcut Tiles on Wear OS by @marazmarci in https://github.com/home-assistant/android/pull/3697
* Allow binary sensors and sensors to be added to driving favorites by @dshokouhi in https://github.com/home-assistant/android/pull/3732
* Allow selecting unavailable entities for complications/tiles by @jpelgrom in https://github.com/home-assistant/android/pull/3733
* Apply various IDE code quality suggestions by @jpelgrom in https://github.com/home-assistant/android/pull/3734
* Invalidate Auto home on favorite state change by @jpelgrom in https://github.com/home-assistant/android/pull/3736
* Hide some sensors from unsupported devices by @dshokouhi in https://github.com/home-assistant/android/pull/3740
* Bump io.sentry:sentry-android from 6.26.0 to 6.27.0 by @dependabot in https://github.com/home-assistant/android/pull/3743
* Initial work on migrating plugins to catalogs. by @JBassett in https://github.com/home-assistant/android/pull/3737
* Migrate deploys to Fastlane by @JBassett in https://github.com/home-assistant/android/pull/3745
* Ignore ruby files that were generated in CI. by @JBassett in https://github.com/home-assistant/android/pull/3746
* Hide hidden entities from android auto domains by @dshokouhi in https://github.com/home-assistant/android/pull/3731
* Handle empty array/invalid URL during discovery by @jpelgrom in https://github.com/home-assistant/android/pull/3747
## New Contributors
* @drosoCode made their first contribution in https://github.com/home-assistant/android/pull/3399
* @slovdahl made their first contribution in https://github.com/home-assistant/android/pull/3703
**Full Changelog**: https://github.com/home-assistant/android/compare/2023.7.5...2023.8.1-beta.0.16+749e9ba
```

**Review by dshokouhi — APPROVED** [review]

**Review by JBassett — APPROVED** [review]

**Pull request merged**